apps/charterafrica/src/payload/blocks/Consultation.js

```diff
@@ -0,0 +1,119 @@
+import {
+  mapPlaylistLinkToId,
+  mapVideoUrlToId,
+} from "../utils/mapPlaylistLinkToId";
+
+const Consultations = {
+  slug: "consultation",
+  labels: {
+    singular: {
+      en: "Consultation Playlist",
+      fr: "Playlist de consultation",
+      pt: "Consulta Lista de reprodução",
+    },
+    plural: {
+      en: "Consultation Playlists",
+      fr: "Playlist de consultations",
+      pt: "Lista de reprodução de consulta",
+    },
+  },
+  fields: [
+    {
+      type: "collapsible",
+      label: {
+        en: "Playlist",
+        fr: "Playlist",
+        pt: "Lista de reprodução",
+      },
+      fields: [
+        {
+          name: "title",
+          label: {
+            en: "Title",
+            fr: "Titre",
+            pt: "Título",
+          },
+          type: "text",
+          required: true,
+          localized: true,
+        },
+        {
+          name: "playlistLink",
+          label: {
+            en: "Playlist",
+            fr: "Playlist",
+            pt: "Lista de reprodução",
+          },
+          type: "text",
+          required: true,
+          localized: true,
+        },
+        {
+          name: "playlistId",
+          label: {
+            en: "Playlist ID",
+            fr: "Identifiant de la liste de lecture",
+            pt: "ID da lista de reprodução",
+          },
+          admin: {
+            hidden: true,
+          },
+          hooks: {
+            beforeValidate: [mapPlaylistLinkToId],
+          },
+          type: "text",
+          localized: true,
+        },
+      ],
+    },
```

Comment:
Playlist content should be a group field (title e.g. Previous Consultations and URL i.e playlist URL) You can use help text to give an example of expected playlist URL.

This way we'll refer to it as playlist.url, playlist.title, etc.